src/Shared/FileUtilities.cs

```diff
@@ -637,19 +637,14 @@ internal static bool PathIsInvalid(string path)
                 return true;
             }
 
-            var filename = GetFileName(path);
+            // Path.GetFileName does not react well to malformed filenames.
+            // For example, Path.GetFileName("a/b/foo:bar") returns bar instead of foo:bar
+            // It also throws exceptions on illegal path characters
+            var lastDirectorySeparator = path.LastIndexOfAny(Slashes);
 
-            return filename.IndexOfAny(InvalidFileNameChars) >= 0;
+            return path.IndexOfAny(InvalidFileNameChars, lastDirectorySeparator > 0 ? lastDirectorySeparator : 0) >= 0;
```

Comment:
I think it needs to be `lastDirectorySeparator + 1`. Otherwise the IndexOfAny search will include the `lastDirectorySeparator`, and thus any path would look invalid: https://dotnetfiddle.net/vNW9TD